src/jit/morph.cpp

```diff
@@ -2953,23 +2951,19 @@ GenTreeCall* Compiler::fgMorphArgs(GenTreeCall* callNode)
         // this can't be a struct.
         assert(argx->gtType != TYP_STRUCT);
 
-        // FIXME: Issue #4025 Why do we need floating type for 'this' argument
```

Comment:
I changed the else part of this to be an uncondional assert stating that the this pointer can not be a floating point type.

That's why I deleted this comment